src/common/options/rgw.yaml.in

```diff
@@ -3236,6 +3236,33 @@ options:
   see_also:
   - rgw_max_objs_per_shard
   - rgw_max_dynamic_shards
+- name: rgw_dynamic_resharding_may_reduce
+  type: bool
+  level: advanced
+  desc: Whether dynamic resharding can reduce the number of shards
+  long_desc: If true, RGW's dynamic resharding ability is allowed to
+    reduce the number of shards if it appears there are too many.
+  default: false
```

Comment:
Where do we sit with changes in default behaviour between releases? Asking because when users upgrade their clusters, this feature would be disabled by default, which is preferable, IMO - that way clusters will continue to behave as they did prior to the upgrade, and users can opt in if they choose to by enabling the new feature.